### PR TITLE
Support building centos-rootfs on Arm64

### PR DESCRIPTION
**Pull request by Pennyzct**
For now, mirrorlist doesn't support non-x86_64 arch, so we need
create baseurl for arm64. Furthermore, we also need to offer
arm64-specific gpg keys along with the regular key.

Fixes: #111

Signed-off-by: Penny Zheng <penny.zheng@arm.com>

### CODE REVIEW

**Comment by Pennyzct** [comment]
@jodh-intel @bergwolf

**Comment by jodh-intel** [comment]
Hi @Pennyzct - looks good. The CI is failing due to github connectivity issues but also you need to add a "subsystem" label to the start of your summary line in your commit (I'd add `rootfs-builder:` maybe). See:

- https://github.com/kata-containers/community/blob/master/CONTRIBUTING.md#patch-format

**Comment by Pennyzct** [comment]
@jodh-intel sorry, refine it asap.

File: rootfs-builder/centos/config_aarch64.sh [hunk 1]
```diff
@@ -0,0 +1,11 @@
+# Base Repos
```
Comment (jcvenegas) [inline rootfs-builder/centos/config_aarch64.sh]: Could you add a header to this file?
```
#
# Copyright (c) 2018 <your-organization> 
#
# SPDX-License-Identifier: Apache-2.0
```

**Comment by jcvenegas** [comment]
@jodh-intel I see licence checker requesting headers for RPM-GPG-KEY-CentOS-7-aarch64 , is there a way to skip it?

**Comment by jodh-intel** [comment]
@jcvenegas - yeah, our checker is a little aggressive :) I guess that's potentially a good thing but annoying at times :smile:

Ideally, we would `curl https://... ` these keys to avoid having to store them locally. If that doesn't work, I guess we'll have to update https://github.com/kata-containers/tests/blob/master/.ci/static-checks.sh to skip such files.

**Comment by jcvenegas** [comment]
@Pennyzct @jodh-intel we can try to do that, 
So we should add an variable (optional) in the rootfs config.
```bash
GPG_KEY_URL=https://www.centos.org/keys/RPM-GPG-KEY-CentOS-7
```
Then in `scripts/lib.sh`
```bash
if [ -n "$GPG_KEY_URL" ]; then
     curl -L GPG_KEY_URL -o gpg-key
     GPG_KEY_FILE=gpg-key
fi
```
@Pennyzct is there a URL were we can curl this new GPG_FILE?

File: rootfs-builder/centos/config_aarch64.sh (hunk 1 above)
Comment (Pennyzct) [reply rootfs-builder/centos/config_aarch64.sh]: my neglect🙂, I will add it asap.

**Comment by Pennyzct** [comment]
Hi~ @jcvenegas. yes, I also download it from `centos.org`. `curl ` would be an better option, and I will do all refinement asap.

**Comment by Pennyzct** [comment]
Hi~ @jcvenegas @jodh-intel  Since `gpgkey` could directly use URL, could we just import url info, not generating related file😊?

**Comment by jodh-intel** [comment]
Hi @Pennyzct - as long as the file can be downloaded securely using `https`. CentOS seems to be fine:

```
$ curl -L --proto \=https --proto-redir \=https -O https://www.centos.org/keys/RPM-GPG-KEY-CentOS-7
```
... but EulerOS doesn't appear to offer `https` download for its key (http://developer.huawei.com/ict/site-euleros/euleros/repo/yum/2.2/os/RPM-GPG-KEY-EulerOS) so it's key file will have to be stored in this repo until that issue is resolved.

/cc @liangchenye, @WeiZhang555.

**Comment by Pennyzct** [comment]
Hi~@jodh-intel thanks for explanation. then i will stick with the `curl` scheme.

**Comment by jodh-intel** [comment]
lgtm

**Comment by jcvenegas** [comment]
lgtm

**Pull request merged**